File: tests/xplat/core/FsharpUnitTestProject/AndroidFSharpLanguageUnitTests/MainActivity.fs

namespace AndroidFSharpLanguageUnitTests

open System
open System.Reflection
open System.Collections.Generic
open System.Linq
open System.Text

open Android.App
open Android.Content
open Android.OS
open Android.Runtime
open Android.Views
open Android.Widget
open Xamarin.Android.NUnitLite
open NUnit.Framework

[<Activity (Label = "AndroidFSharpLanguageUnitTests", MainLauncher = true)>]
type MainActivity() =
  inherit TestSuiteActivity()

    override x.OnCreate(bundle) =
        x.AddTest(Assembly.GetExecutingAssembly())
        base.OnCreate (bundle)

[<TestFixture; Description("Blah blah")>]
type ``My tests``() =
    [<Test>]
    member x.``Subtype Tests``() =
        let results = true
        Assert.IsTrue(results)

File: tests/xplat/core/FsharpUnitTestProject/AndroidWrapper/MainActivity.cs
using System.Reflection;
using Android.App;
using Android.OS;
using Xamarin.Android.NUnitLite;

namespace CUnitTests
{
    [Activity (Label = "CSUnitTests", MainLauncher = true)]
    public class MainActivity : TestSuiteActivity
    {
        protected override void OnCreate (Bundle bundle)
        {
            // tests can be inside the main assembly
	    AddTest (typeof(Core_access).Assembly );
            // or in any reference assemblies
            // AddTest (typeof (Your.Library.TestClass).Assembly);

            // Once you called base.OnCreate(), you cannot add more assemblies.
            base.OnCreate (bundle);
        }
    }
}

